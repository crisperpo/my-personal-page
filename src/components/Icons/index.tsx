
import {
    GithubLogo as GithubIcon,
    LinkedinLogo as LinkedinIcon,
    MediumLogo as MediumIcon,
    PinterestLogo as PinterestIcon
} from '@phosphor-icons/react';
import BlueskyIcon from './BlueskyIcon';

export default { BlueskyIcon, GithubIcon, LinkedinIcon, MediumIcon, PinterestIcon };
