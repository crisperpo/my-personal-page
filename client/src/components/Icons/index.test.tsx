import { render } from '@testing-library/react';
import Icons from './index';

describe('Icons Component', () => {
    it('should match the snapshot for BlueskyIcon', () => {
        const { container } = render(<Icons.BlueskyIcon />);
        expect(container).toMatchSnapshot();
    });

    it('should render BlueskyIcon correctly', () => {
        const { container } = render(<Icons.BlueskyIcon />);
        expect(container).toMatchSnapshot();
    });

    it('should render GithubIcon correctly', () => {
        const { container } = render(<Icons.GithubIcon />);
        expect(container).toBeInTheDocument();
    });

    it('should render LinkedinIcon correctly', () => {
        const { container } = render(<Icons.LinkedinIcon />);
        expect(container).toBeInTheDocument();
    });

    it('should render MediumIcon correctly', () => {
        const { container } = render(<Icons.MediumIcon />);
        expect(container).toBeInTheDocument();
    });

    it('should render PinterestIcon correctly', () => {
        const { container } = render(<Icons.PinterestIcon />);
        expect(container).toBeInTheDocument();
    });
});
