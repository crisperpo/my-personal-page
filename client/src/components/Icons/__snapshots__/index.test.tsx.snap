// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icons Component > should match the snapshot for BlueskyIcon 1`] = `
<div>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  />
</div>
`;

exports[`Icons Component > should render BlueskyIcon correctly 1`] = `
<div>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  />
</div>
`;
